sympy/logic/algorithms/z3_wrapper.py

```diff
@@ -29,11 +29,16 @@ def z3_satisfiable(expr, all_models=False):
     if res == "unsat":
         return False
     elif res == "sat":
-        return True
+        return z3_model_to_sympy_model(s.model(), expr)
     else:
         return None
 
 
+def z3_model_to_sympy_model(z3_model, enc_cnf):
+    rev_enc = {value : key for key, value in enc_cnf.encoding.items()}
+    return {rev_enc[int(var.name()[1:])] : bool(z3_model[var]) for var in z3_model}
```

Comment:
The `bool` function is being used here to convert a z3 [BoolRef](https://z3prover.github.io/api/html/classz3py_1_1_bool_ref.html) object into a python True or False. I think it might be better to do this in a different way because if for some reason there was a bug and z3_model[var] was some other object, bool might return True and it would be hard to tell that there was a bug.

Reply:
`var` is always going to be some single character name here?

Reply:
> The bool function is being used here to convert a z3 [BoolRef](https://z3prover.github.io/api/html/classz3py_1_1_bool_ref.html) object into a python True or False. I think it might be better to do this in a different way because if for some reason there was a bug and z3_model[var] was some other object, bool might return True and it would be hard to tell that there was a bug.

You can change this if you want, but I also think this way is fine.